src/main/java/com/socialmedia/repository/UserRepository.java

```diff
@@ -0,0 +1,10 @@
+package com.socialmedia.repository;
+
+import com.socialmedia.model.ApplicationUser;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.Optional;
+
+public interface UserRepository extends JpaRepository<ApplicationUser, String> {
+  Optional<ApplicationUser> findByUsername(String  username);
```

Comment:
Jpa repository already contains method findById. As username is id for ApplicationUser entity, this method may be removed and instead you should use findById method

Reply:
done